File: taurus-protect-sdk-python/tests/testutil/properties.py
"""Simple key=value properties file parser."""

from __future__ import annotations

from typing import Dict


def load_properties(path: str) -> Dict[str, str]:
    """Load a Java-style .properties file.

    Supports:
    - key=value pairs
    - # comments
    - Blank lines (ignored)
    - \\n escape sequences in values (for PEM keys)

    Args:
        path: Path to the properties file.

    Returns:
        Dictionary of key-value pairs.
    """
    props: Dict[str, str] = {}
    try:
        with open(path, "r") as f:
            for line in f:
                line = line.strip()
                if not line or line.startswith("#"):
                    continue
                eq_idx = line.find("=")
                if eq_idx < 0:
                    continue
                key = line[:eq_idx].strip()
                value = line[eq_idx + 1:].strip()
                # Unescape \n in PEM keys
                value = value.replace("\\n", "\n")
                props[key] = value
    except FileNotFoundError:
        pass
    return props

File: taurus-protect-sdk-typescript/tests/testutil/properties.ts
/**
 * Simple key=value properties file parser.
 *
 * Supports:
 * - Lines with `key=value` format
 * - `#` comment lines (ignored)
 * - Blank lines (ignored)
 * - `\n` escape sequences in values (for PEM keys)
 */

import * as fs from 'fs';

/**
 * Parses a properties file into a key-value map.
 *
 * @param filePath - Path to the properties file
 * @returns Map of property keys to values, or null if file not found
 */
export function loadProperties(filePath: string): Map<string, string> | null {
  let content: string;
  try {
    content = fs.readFileSync(filePath, 'utf-8');
  } catch {
    return null;
  }

  const props = new Map<string, string>();
  const lines = content.split('\n');

  for (const line of lines) {
    const trimmed = line.trim();
    if (trimmed === '' || trimmed.startsWith('#')) {
      continue;
    }

    const eqIndex = trimmed.indexOf('=');
    if (eqIndex < 0) {
      continue;
    }

    const key = trimmed.substring(0, eqIndex).trim();
    let value = trimmed.substring(eqIndex + 1).trim();

    // Unescape \n in values (e.g., PEM keys)
    value = value.replace(/\\n/g, '\n');

    if (key) {
      props.set(key, value);
    }
  }

  return props;
}
